packages/colorpickers/src/elements/ColorPicker/ColorWell.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useRef, useState, useContext } from 'react';
+import { ThemeContext } from 'styled-components';
+import throttle from 'lodash.throttle';
+import { IHSVColor } from './reducer';
+import { hsl2hsv } from '../../utils/conversion';
+import { calculateNextHsv, getSaturationPosition } from '../../utils/saturation';
+import {
+  StyledColorWell,
+  StyledColorWellGradient,
+  StyledColorWellThumb,
+  StyledColorWellWrapper
+} from '../../styled';
+
+interface IColorWellProps {
+  hue: number;
+  saturation: number;
+  lightness: number;
+  onChange?: (hsv: IHSVColor, event: React.MouseEvent<any>) => void;
+}
+
+export const ColorWell: React.FC<IColorWellProps> = ({ hue, saturation, lightness, onChange }) => {
+  const { rtl } = useContext(ThemeContext);
+  const container = useRef<HTMLDivElement>(null);
+  const hsv = hsl2hsv(hue, saturation, lightness);
+  const [position, setPosition] = useState<{ s: number; v: number }>({
+    s: rtl ? 100 - hsv.s : hsv.s,
+    v: hsv.v
+  });
+
+  const throttledChange = throttle(e => {
+    if (container.current) {
+      const nextHsv = calculateNextHsv(e, hsv, container.current, rtl);
+      const saturationPosition = getSaturationPosition(e, container.current);
+
+      setPosition(saturationPosition);
+      onChange && onChange(nextHsv, e);
+    }
+  }, 50);
+
+  const topPosition = 100 - position.v;
+  const leftPosition = position.s;
+
+  const handleChange = (e: any) => throttledChange(e);
+
+  const handleMouseUp = () => {
+    throttledChange.cancel();
+    window.removeEventListener('mousemove', handleChange);
+    window.removeEventListener('mouseup', handleMouseUp);
```

Comment:
If we're attaching global events we need a useEffect to remove these in the event the component is unmounted before the mouseUp event occurs.

Reply:
You could also have a single `isMousedDown` state with an effect that applies/removes events based on that value.

Reply:
In what scenario would `ColorWell` unmount prior to the `mouseup` event? I'm unable to think of such a scenario. Would there be bigger issues than lingering event listeners in such a scenario?

Reply:
It would be rare but is still a possibility. Unmounts can be triggered by most things outside of our control. 

In general, the best practice is to remove global event listeners within a useEffect, right? [MultiThumbRange example](https://github.com/zendeskgarden/react-components/blob/main/packages/forms/src/elements/MultiThumbRange.tsx#L94-L102)

Reply:
The `MultiThumbRange` code aligns with how I think about registering and cleaning up listeners in a `useEffect`. I think it's best practice to remove event listeners registered within a `useEffect` in its cleanup function. 

However, these event listeners aren't registered inside a `useEffect`. It seems fair to register the listeners on `mousedown` and remove them on the corresponding `mouseup` event.